### PR TITLE
project-movies: Maria Pogosjan & Maidelin Rubio

### PR DESCRIPTION
**Pull request by MariaPogosjan**
Link to my deployed project:

https://movie-netflix.netlify.app/

My reflections on how this week's project turned out:
It was a fun project and good to understand how to create multiple pages. 

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/index.css [hunk 1]
```diff
@@ -1,13 +1,216 @@
...
+}
+
+.movie-title {
+  margin-top: auto;
+  margin-bottom: 0;
+  font-size: 35px;
+}
+
+/*** SECOND PAGE STYLING ***/
+.detail-background {
```
Comment (FD000) [inline code/src/index.css:72]: When you are re-sizing the screen, the background picture stretches and is not responsive. Try using `object-fit: cover ` as styling to this picture and everything will work fine.

File: code/src/index.css [hunk 2]
```diff
@@ -1,13 +1,216 @@
...
+    width: 100%;
+    height: 100%;
+    display: flex;
+    justify-content: flex-start;
+    align-items: center;
+    flex-wrap: wrap;
+    padding: 50px;
+  }
+
+  .detail-content {
```
Comment (FD000) [inline code/src/index.css:197]: Maybe you could add a detail semi-transparent background to separate text from the backdrop, which will increase readability.

File: code/src/index.css [hunk 3]
```diff
@@ -1,13 +1,216 @@
...
+    width: 28%;
+    display: flex;
+    flex-direction: column;
+    margin-left: 50px;
+  }
+ 
+}
+
+@media (min-width: 1200px) {
```
Comment (FD000) [inline code/src/index.css:206]: image 
The movie title and the ratings are two separate boxes.  Maybe we could instead have them in same box and create a span after the movie title and then add the rating there to make sure it follows the title as the text wraps.

File: code/src/components/Loading.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import Loader from 'react-loader-spinner'
+import "react-loader-spinner/dist/loader/css/react-spinner-loader.css"
+
+
+const Loading =()=>{
+  return (
+    <div>
+      <Loader
```
Comment (FD000) [inline code/src/components/Loading.js:9]: Fantastic, very beautiful loader!

File: code/src/index.css [hunk 5]
```diff
@@ -1,13 +1,216 @@
...
+ font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Fira Sans,Droid Sans,Helvetica Neue,sans-serif;
+    -webkit-font-smoothing: antialiased;
+}
+
+/*** FIRST PAGE STYLING ***/
+.filter-btn-container {
+  height: 50px;
+}
+
+.filter-btn {
```
Comment (FD000) [inline code/src/index.css:18]: Beautiful buttons when changing movie filter. Its easy and smooth to change category

File: code/public/index.html [hunk 6]
```diff
@@ -4,6 +4,7 @@
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/png" href="./favicon/popcorn.png">
```
Comment (FD000) [inline code/public/index.html:7]: Very beautiful popcorn favicon, we really liked it :)

File: code/src/pages/Detail.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        <Route path="/">
+          <Link className="home-page-link" to={"/"}>
+            <img src="../favicon/button.svg" alt="button icon" />
+            <h4 className="link-title">Movies</h4>
+          </Link>
+        </Route>
+        {/* Link back to similar movie list */}
+        <Route path="/movie/:id">
+          <Link className="other-page-link" to= {`/similar/${params.id}`}>
+            <img src="../favicon/button.svg" alt="button icom" />
```
Comment (FD000) [inline code/src/pages/Detail.js:69]: Maybe have the Alternative TItles pointer to point towards right and not left like the Movies pointer above.

**Review by FD000 — COMMENTED** [review]
Very beautiful app. We really liked the styiling. All that remains is some responsiveness for mobile viewports. Keep up the good work!
// Team DD, Martin D & Fethullah D

File: code/src/pages/MovieList.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import MovieElement from '../components/MovieElement'
+
+const MovieList = ({filterType}) => {
+  const [movieList, setMovieList] = useState([])
+
+  useEffect (() => {
+   fetch(API_URL(filterType))
+      .then(response => response.json())
+      .then(data => setMovieList(data.results))
+      .catch (err => console.dirr(err))
```
Comment (KarolCislowski) [inline code/src/pages/MovieList.js:14]: Please remember to remove all console outputs from your code before submitting. 👮🏻

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Nice project but please remove all console outputs.

To easily create 404 page you need to:
1. throw en error inside of `.then` block,
2. catch it in the `.catch` block and redirect it with `useHistory` hook provided by `React-router`
3. create a separate page(component) where the user will be redirected.
4. extra step if you are using netlify to host your app: create inside the `public` folder the `_redirects` file and save this line in it: `/* /index.html 200`

Try it 🦄

**Comment by MariaPogosjan** [comment]
Thank you for your feedback! The error message is now fixed :)


*Allt gott, *
*Maria *

----------

Maria Pogosjan
Phone

|   +46 (0)73 69 14 411  <http://sdf/>




Den lör 10 apr. 2021 kl 19:25 skrev Karol Cislowski <
***@***.***>:

**Review by KarolCislowski — APPROVED** [review]

**Pull request closed without merging**